flink/flink-common/src/main/java/com/netease/arctic/flink/FlinkSchemaUtil.java

```diff
@@ -197,4 +245,122 @@ public static void addPrimaryKey(
                               "Arctic primary key should be declared in table")));
         });
   }
+  /**
+   * Generate table properties for watermark and computed columns from flink TableSchema.
+   *
+   * @param schema flink TableSchema.
+   * @return Table properties map.
+   */
+  public static Map<String, String> addSchemaProperties(TableSchema schema) {
+    Map<String, String> properties = Maps.newHashMap();
+
+    // field name -> index
+    final Map<String, Integer> indexMap = new HashMap<>();
+    List<TableColumn> tableColumns = schema.getTableColumns();
+    for (int i = 0; i < tableColumns.size(); i++) {
+      indexMap.put(tableColumns.get(i).getName(), i);
+    }
+
+    List<TableColumn> computedColumns =
+        schema.getTableColumns().stream()
+            .filter(column -> column instanceof ComputedColumn)
+            .collect(Collectors.toList());
+    properties.putAll(serializeComputeColumn(indexMap, computedColumns));
+
+    // watermark,only support one watermark now
+    List<WatermarkSpec> watermarkSpecs = schema.getWatermarkSpecs();
+    if (!watermarkSpecs.isEmpty()) {
+      if (watermarkSpecs.size() > 1) {
+        throw new IllegalStateException("Multiple watermark definition is not supported yet.");
+      }
+      properties.putAll(serializeWatermarkSpec(watermarkSpecs.get(0)));
+    }
+
+    return properties;
+  }
+
+  /** serialize computeColumns into properties */
+  private static Map<String, String> serializeComputeColumn(
+      Map<String, Integer> indexMap, List<TableColumn> computedColumns) {
+    Map<String, String> serialized = new HashMap<>();
+    computedColumns.stream()
+        .forEach(
+            column -> {
+              int index = indexMap.get(column.getName());
+              serialized.put(
+                  compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, index, NAME), column.getName());
+              serialized.put(
+                  compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, index, DATA_TYPE),
+                  column.getType().getLogicalType().asSerializableString());
+              serialized.put(
+                  compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, index, EXPR),
+                  ((TableColumn.ComputedColumn) column).getExpression());
+            });
+
+    return serialized;
+  }
+
+  /** deserialize computeColumns from properties */
+  private static TableColumn deserializeComputeColumn(
+      Map<String, String> arcticProperties, int index) {
+    String name = arcticProperties.get(compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, index, NAME));
```

Comment:
I think we should check if the column name exists in the Amoro Schema. 
If the column name has been renamed, but the options didn't update. We should throw an exception when meeting this case.
And we should add a corner test for it.

Reply:
> I think we should check if the column name exists in the Amoro Schema. If the column name has been renamed, but the table options weren't updated. We should throw an exception when meeting this case. And we should add a corner test for it.

@YesOrNo828  I think it's necessary.
 Do you mean the phenomenon below?
 1. Flink DDL  
physical colum id and compute column cal_id
 ```
CREATE TABLE mix_iceberg.test_db.test_db (  
        `id` int,          
        `cal_id` as id+5,        
        proc as PROCTIME()       
        )  
```
tableProperties as below
```
flink.computed-column.1.expr= `id` + 5
flink.computed-column.1.name= cal_id
flink.computed-column.1.data-type= INT
```

2. other engine, such spark rename `id` into `newid`
`ALTER TABLE mix_iceberg.test_db.test_db RENAME COLUMN id TO newid;`

3. deserializeComputeColumn function should throw exception if tableProperties weren't updated for `flink.computed-column.1.xx`
4. deserializeComputeColumn function work normaly, after column renamed and tableProperties are updated